### PR TITLE
backport, bugfix and new mate-menus api support

### PR DESCRIPTION
**Pull request by yetist**
Before that, I submitted a [PR](https://github.com/mate-desktop/mate-menus/pull/59) for mate-menus, if that PR merged, it will break mozo.

So now I've backport the alacarte code to mozo, and now it works good and some bug has been fixed, but more review and tests are needed, because this is a huge change.

### CODE REVIEW

**Comment by raveit65** [comment]
Without agreeing to this re-fork, translations are handle by transifex, so no need to change .po files
We submit changes to transifex server and the .po files will generate there after translations are done and we download changes before we release a new tarball.
Are you familiar 'with the 'git rebase -i last-good-commit-id' to change your commits?

**Comment by yetist** [comment]
I also updated the .po files, that include the new msgid from ui files, and some po has already translated, see [here](https://github.com/mate-desktop/mozo/pull/39/commits/75bbee3513c2eaf0f5ced44e5540aa2d33f3729d). 

If you don't like the translation with PR, I can revert it, but the translator may have to work more.

**Comment by raveit65** [comment]
All, you changes will be overwritten if we generate a new resource mozo.pot for transifex.
So it is useless...

**Comment by lukefromdc** [comment]
We do need to remember that this version of the Mozo PR requires the matching version of the mate-menus PR.

File: mozo.in [hunk 1]
```diff
@@ -28,7 +28,7 @@ def main():
         version = config.VERSION
     except:
         datadir = './data/'
-        version = '1.7.0'
+        version = '1.21.0'
```
Comment (monsta) [inline mozo.in]: This is weird... no, not your changes. I mean, why do we have some fallback here? If we can't import `config`, the package is damaged, and we should just quit.

**Comment by monsta** [comment]
Hmm... what's the point of new `LauncherEditor` and `DirectoryEditor`?

File: configure.ac [hunk 2]
```diff
@@ -18,9 +18,9 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package])
 AM_GLIB_GNU_GETTEXT
 IT_PROG_INTLTOOL([0.40.0])
 
-AM_PATH_PYTHON_VERSION(2.7, 2.7.0)
+AM_PATH_PYTHON_VERSION(3.6, 3.6.0, 3.5, 3.5.0, 2.7, 2.7.0)
```
Comment (monsta) [inline configure.ac]: Version 2.7 shouldn't be valid anymore after porting to Python 3.

I have a feeling we should just drop the whole `acinclude.m4` which contains `AM_PATH_PYTHON_VERSION`, and just use `AM_PATH_PYTHON`.

**Comment by raveit65** [comment]
Looks like we have open questions here?

**Comment by monsta** [comment]
We also still have version bumps here and in mate-menus.

**Comment by monsta** [comment]
Wonder why we need a new `mozo.svg` image here...

**Comment by raveit65** [comment]
@yetist 
Can you please revert version bump?
Last commit can be squashed?

**Comment by yetist** [comment]
@raveit65 
>Can you please revert version bump?
> Last commit can be squashed?

This is done.

File: Mozo/MenuEditor.py [hunk 3]
```diff
@@ -22,8 +22,7 @@
 import xml.dom.minidom
 import xml.parsers.expat
 import locale
-import matemenu
-from gi.repository import GLib
+from gi.repository import MateMenu, GLib
```
Comment (monsta) [inline Mozo/MenuEditor.py]: Maybe add `gi.require_version` here...

**Comment by monsta** [comment]
Ok, asked for review from developers who probably know Python better than me :slightly_smiling_face:

**Comment by raveit65** [comment]
> Ok, asked for review from developers who probably know Python better than me :slightly_smiling_face: 

@vkareh @XRevan86 
Can you please take a look at it ?

**Comment by monsta** [comment]
Some parts of this can be squashed. For example, `LauncherEditor` and `DirectoryEditor` were first added then removed.

**Comment by lukefromdc** [comment]
mate-menus is now merged, so this needs to be merged as well

**Comment by raveit65** [comment]
> Some parts of this can be squashed. For example, LauncherEditor and DirectoryEditor were first added then removed.

Looks like this is gone, a 'grep' on the repo gives me no result.
Do we have gold status?

**Comment by monsta** [comment]
Ok, it's already squashed into one commit.

**Comment by raveit65** [comment]
Hmm, why always 1 commit or two much commits?
Why not taken the middle? Lets say 5-6 for such a big change.

**Pull request merged**